storage/coin_storage_test.go

```diff
@@ -373,6 +447,66 @@ func TestCoinStorage(t *testing.T) {
 		assert.Equal(t, &types.CoinIdentifier{Identifier: "coin4"}, coinIdentifier)
 		assert.Equal(t, blockIdentifier, block)
 	})
+
+	t.Run("AddCoins after block", func(t *testing.T) {
+		accountCoins := []*AccountCoin{
+			&AccountCoin{
+				Account: account,
+				Coin:    coins1,
+			},
+			&AccountCoin{
+				Account: account2,
+				Coin:    coins2,
+			},
+			&AccountCoin{
+				Account: account3,
+				Coin:    coins3,
+			},
+			&AccountCoin{
+				Account: account4,
+				Coin:    coins4,
+			},
+		}
+
+		account1coins, block, err := c.GetCoins(ctx, account)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+
+		account2coins, block, err := c.GetCoins(ctx, account2)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+
+		account3coins, block, err := c.GetCoins(ctx, account3)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+
+		err = c.AddCoins(ctx, accountCoins)
+		assert.NoError(t, err)
+
+		coinsGot, block, err := c.GetCoins(ctx, account)
+		account1coins = append(account1coins, coins1)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+		assert.ElementsMatch(t, coinsGot, account1coins)
+
+		coinsGot, block, err = c.GetCoins(ctx, account2)
+		account2coins = append(account2coins, coins2)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+		assert.ElementsMatch(t, coinsGot, account2coins)
+
+		coinsGot, block, err = c.GetCoins(ctx, account3)
+		account3coins = append(account3coins, coins3)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+		assert.ElementsMatch(t, coinsGot, account3coins)
+
+		// Does not add duplicate
+		coinsGot, block, err = c.GetCoins(ctx, account4)
+		assert.NoError(t, err)
+		assert.Equal(t, blockIdentifier, block)
+		assert.ElementsMatch(t, coinsGot, []*types.Coin{coins4})
+	})
 }
 
 type MockCoinStorageHelper struct{}
```

Comment:
realized my mocking here is sub-par...you should allow for current block identifier to return an ErrHeadBlockNotFound here when adding coins (right now we always return a block as if we had synced)